automation/src/services/element-handler.ts

```diff
@@ -0,0 +1,78 @@
+import {ElementHandle, Page} from 'puppeteer-core';
+
+/**
+ * <pre>
+ * Get the value of a particular property for a particular element
+ * </pre>
+ * @param {Page} page
+ * @param {ElementHandle} element - The web element to get the property value for
+ * @param {string} property - The property to look for
+ * @returns {string} value - The property value
+ *
+ */
+export async function getProperty(page: Page, element: ElementHandle, property: string) {
+  // Alternative: return element.getProperty(property).then((elem) => elem.jsonValue());
+  return page.evaluateHandle((elem, prop) => {
+    return elem[prop];
+  }, element, property).then((handle) => handle.jsonValue());
+}
+
+/**
+ * Get the element attribute value.
+ * @param {Page} page
+ * @param {ElementHandle} element
+ * @param {string} attribute
+ */
+export async function getAttribute(page: Page, element: ElementHandle, attribute: string) {
+  return page.evaluateHandle((elem, attr) => {
+    return elem.getAttribute(attr);
+  }, element, attribute).then((handle) => handle.jsonValue());
+}
+
+/**
+ * <pre>
+ * Get the textContent property value for a element.
+ * </pre>
+ * @param {Page} page
+ * @param {ElementHandle} element - The web element
+ * @returns {string} value - The text property value
+ */
+export async function getText(page: Page, element: ElementHandle) {
+  return page.evaluate(elem => elem.textContent, element);
+}
+
+/**
+ * Get the value property value for a element.
+ * @param page
+ * @param element
+ */
+export async function getValue(page: Page, element: ElementHandle) {
+  // Alternative: await (await elem.getProperty('value')).jsonValue();
+  return page.evaluate(elem => elem.innerText, element);
+}
+
+/**
+ * <pre>
+ * Check whether a web element exist. Visibility is not checked.
+ * </pre>
+ * @param {Page} page
+ * @param {ElementHandle} element: The web element to check
+ */
+export async function exist(page: Page, element: ElementHandle) {
+  return page.evaluate(elem => {
+    return elem !== null;
+  }, element);
+}
+
+/**
+ * <pre>
+ *  Check if the element is visible
+ * </pre>
+ * @param {Page} page
+ * @param {ElementHandle} element
+ */
+export async function visible(page: Page, element: ElementHandle) {
```

Comment:
naming nitpick: isVisible? It just feels more clearly like an assertion type method that way.